### PR TITLE
Pbi/93611 persist post put to indexed db

### PR DESCRIPTION
**Pull request by arildbjerkenes**
Denne PR-en inneholder både persist av updates, restrukturering og oppdatering av data for new punch. 

Det ligger noe log her og der, som tas bort senere. Det blir sikkert en runde til med omstrukturering.

### CODE REVIEW

**Review by VRiveland — CHANGES_REQUESTED** [review]
I can't seem to be able to create a new punch while offline. There doesn't seem to exist a table for  saving post/put/delete in indexedDB

File: src/offline/offlineContentUpdates/handleNewPunch.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        formularType: '',
+        id: newPunchItemId,
+        isRestrictedForUser: false, //??
+        materialEta: null,
+        materialNo: null,
+        materialRequired: false, //??
+        priorityCode: null,
+        priorityDescription: null,
+        priorityId: null,
+        raisedByCode: '',
```
[the remark this answers is not in the document]
Comment (arildbjerkenes) [reply src/offline/offlineContentUpdates/handleNewPunch.ts:79]: Seems that in the fetch request we have RaisedByOrganizationId, but in the response we have RaisedByCode. Should we include RaisedByCode in the response from the endpoint?

File: src/offline/offlineContentUpdates/handleNewPunch.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        actionByPerson: newPunch.ActionByPerson,
+        actionByPersonFirstName: null,
+        actionByPersonLastName: null,
+        attachmentCount: 0,
+        checklistId: newPunch.CheckListId,
+        clearedAt: null,
+        clearedByFirstName: null,
+        clearedByLastName: null,
+        clearedByUser: null,
+        clearingByCode: '',
```
[the remark this answers is not in the document]
Comment (arildbjerkenes) [reply src/offline/offlineContentUpdates/handleNewPunch.ts:65]: Same comment as above. We have ClearingByOrganizationId.

File: src/offline/offlineContentUpdates/handleNewPunch.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (VRiveland) [reply src/offline/offlineContentUpdates/handleNewPunch.ts:79]: We already have the code that goes with the organzationId, see the PunchOrganization type in ApiTypes. You'll have to handle this ish the same way as matching category id to PA/PB punch type.

[the remark this answers is not in the document]
Comment (arildbjerkenes) [reply src/offline/offlineContentUpdates/handleNewPunch.ts:79]: Ah!

**Review by VRiveland — APPROVED** [review]

**Pull request merged**